tests/structure/test_rings.py

```diff
@@ -188,3 +188,43 @@ def test_no_adjacent_stacking_interactions():
     interactions = struc.find_stacking_interactions(molecule)
 
     assert len(interactions) == 0
+
+
+def test_find_pi_cation_interactions():
+    """
+    Test π-cation interaction detection between aromatic residues and charged ligands.
+    Uses PDB 3wip known to have pi-cation interactions between tryptophan/tyrosine
+    residues and acetylcholine (ACH) ligand.
+    """
+    pdbx_file = pdbx.CIFFile.read(Path(data_dir("structure")) / "3wip.cif")
```

Comment:
Could you perform a quick check if any of the other test structures already fulfills this purpose, if it is not too time-consuming? I am a bit reluctant to add new test structures, if they are not necessary, as every one increases the size of the repo. Or is there another special reason to pick `3wip`?

Reply:
<img width="638" alt="image" src="https://github.com/user-attachments/assets/a564564e-5b4c-4f67-98b0-2d707f3c5752" />

Reply:
Usually 1 out of 4 TRP would  be involved in pi-cation interactions, I tried looking for backbone pi cation interactions between aromatic residues and cationic residues. To check this further I ran a plip batched script to analyse the protein-ligands interactions and none of them yielded any pi-cation interactions. 

The choice of `3wip` was just based on a suggestion from perplexity, there maybe smaller pdb files that contain this interaction I can check for those : ))